frontend/catalyst/qfunc.py

```diff
@@ -89,6 +89,42 @@ def extract_backend_info(
     def __call__(self, *args, **kwargs):
         assert isinstance(self, qml.QNode)
 
+        # Mid-circuit measurement configuration/execution
+        mcm_config = self.execute_kwargs["mcm_config"]
+        _dynamic_one_shot_called = kwargs.pop("_dynamic_one_shot_called", False)
+        total_shots = (
+            self.device.shots
+            if isinstance(self.device, qml.devices.LegacyDevice)
+            else self.device.shots.total_shots
```

Comment:
Are there many places where this could be avoided implementing a `get_total_shots` method in all devices?

Reply:
Probably. There are variations of the same logic across PL as well. I think I'll add a private helper to `qfunc.py` to get the shots and we can replace it if/when a similar helper is added to vanilla PL